Add recipe for just-mode.el

leon-barrett opened the pull request:
### Brief summary of what the package does

A major mode for editing justfiles, as defined by the tool "just": https://github.com/casey/just

### Direct link to the package repository

https://github.com/leon-barrett/just-mode.el

### Your association with the package

I am the maintainer.

### Relevant communications with the upstream package maintainer

None needed.

### Checklist

Please confirm with `x`:

- [X] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses).
- [X] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [X] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [X] My elisp byte-compiles cleanly
- [X] `M-x checkdoc` is happy with my docstrings
- [X] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [ ] I have confirmed some of these without doing them

riscy commented:
Thanks for this.  Mostly looks good, just a few suggestions:

- Is this a debug line that should be removed? `(message "prog-first-column %s" (prog-first-column))`
- You can run <kbd>M-: (lm-commentary)</kbd> in your buffer and preview approximately how your Commentary will appear on MELPA's index -- in this case the "Description:" header will be redundant, and the horizontal rule `;;;;` will appear a little unusual
- just-mode.el#L166: It's safer to sharp-quote function names; use `#'`
- just-mode.el#L109: It's safer to sharp-quote function names; use `#'`
- just-mode.el#L118: It's safer to sharp-quote function names; use `#'`

leon-barrett commented:
Thanks for the review! I have made these changes:

* I removed the debug line (oops).
* I cleaned up the commentary.
* I sharp-quoted four functions.

riscy commented:
Looks good, thanks for this.  One more thing I noticed (non-blocking): could `(= 0 (- (line-end-position) (line-beginning-position)))` be `(and (bolp) (eolp))`

And, welcome to MELPA. :)

leon-barrett commented:
Thanks, I'll do that. And thanks, glad to finally contribute!
